pkg/rpc/server/server.go

```diff
@@ -1873,7 +1889,153 @@ func (s *Server) runScriptInVM(t trigger.Type, script []byte, contractScriptHash
 	if err != nil {
 		faultException = err.Error()
 	}
-	return result.NewInvoke(ic, script, faultException, s.config.MaxIteratorResultItems), nil
+	var registerSession result.OnNewSession
+	if s.config.SessionEnabled {
+		registerSession = s.registerSession
+	}
+	return result.NewInvoke(ic, script, faultException, registerSession, s.config.MaxIteratorResultItems), nil
+}
+
+// registerSession is a callback used to add new iterator session to the sessions list.
+// It performs no check whether sessions are enabled.
+func (s *Server) registerSession(sessionID string, iterators []result.ServerIterator, finalize func()) {
+	s.sessionsLock.Lock()
+	sess := &session{
+		iterators: iterators,
+		finalize:  finalize,
+		till:      time.Now().Add(time.Second * time.Duration(s.config.SessionExpirationTime)),
+	}
+	s.sessions[sessionID] = sess
+	s.sessionsLock.Unlock()
+}
+
+func (s *Server) handleSessions() {
```

Comment:
Do we need another tick-tocking routine? You can do `timer.After()` right in the `registerSession` that will lock/finalize/drop from the list. The only problem is concurrency wrt `Shutdown`, but that can be solved.

Reply:
`timer.After()` returns a channel, so we have to wait until timer fires in a separate routine. New goroutine for each new iterator session is not efficient, it's better to have a single ticker routine IMO. Although maybe there's another way to wait until timer fires (without new routine creation)?

Reply:
`AfterFunc`, sorry. It will run the callback given in a new goroutine, but it won't run for long.

Reply:
Fixed.